### PR TITLE
feat(spell-description): Add hover over spells

### LINKED ISSUE #15: Spell Description
Add a popup box that shows up when you hover over a spell with a description.

### PR DESCRIPTION
**Pull request by felipebrunetti94**
Gravação de tela de 14-11-2019 11_41_14
- Add hover over the menu area
- Still need the spells descriptions

Fix #15

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -61,5 +61,29 @@
 	
 	setupCanvas();
 
-	initSounds();
-</script>
\ No newline at end of file
+  initSounds();
+  
+var gameCanvas = document.getElementById("gameCanvas");
```
Comment (negreirosleo) [inline index.html]: var? :thinking:

Comment (negreirosleo) [reply index.html]: Btw, dont use ID's for selectors, you should use data-attributes instead.

File: js/game.js [hunk 2]
```diff
@@ -56,13 +58,26 @@ function draw() {
...
-			drawText(spellText, 20, false, 120+i*40, "aqua");
-		}
-	}
+    drawSpells()
 }
 
+
+function drawSpells() {
+  
+  for(let i=0; i<player.spells.length; i++){
```
Comment (negreirosleo) [inline js/game.js]: How about a `forEach` here since player.spells is an array?

File: js/game.js [hunk 3]
```diff
@@ -56,13 +58,26 @@ function draw() {
...
-		for(let i=0; i<player.spells.length; i++){
-			let spellText = (i+1) + ") " + (player.spells[i] || "");
-			drawText(spellText, 20, false, 120+i*40, "aqua");
-		}
-	}
+    drawSpells()
 }
 
+
+function drawSpells() {
```
Comment (negreirosleo) [inline js/game.js]: Use arguments 

```js
function drawSpells(player,hoverMenu) {
```

File: js/game.js [hunk 4]
```diff
@@ -56,13 +58,26 @@ function draw() {
...
+  
+  for(let i=0; i<player.spells.length; i++){
+    let spellText = `${i+1}) ${player.spells[i] || "No spells"}`;
+    
+    if(hoverMenu)
+    {
+      drawText(`Description`, 20, false, 120+i*40, "aqua");
```
Comment (negreirosleo) [inline js/game.js]: What does `120+i*40` mean? can this calculation be assigned to a variable?

```js
const calculationMeaning = 120+i*40
```

File: index.html [hunk 5]
```diff
@@ -61,5 +61,29 @@
...
+  const menuLeft = canvasX + gameAreaWidth
+
+  if(mouseX > menuLeft){
+    hoverMenu = true
+  }  else {
+    hoverMenu = false
+  }        
+}
+
+gameCanvas.addEventListener('mousemove', () => mouseTracker(), true)
```
Comment (negreirosleo) [inline index.html:88]: Why is this in the html? how about creating a file for this?

File: index.html (hunk 1 above)
Comment (felipebrunetti94) [reply index.html]: data attributes ? How exactly, could you give me an example ?

File: js/game.js (hunk 2 above)
Comment (felipebrunetti94) [reply js/game.js]: Sure, I forgot about `foreach` kkk

File: js/game.js (hunk 3 above)
Comment (felipebrunetti94) [reply js/game.js]: good one

**Comment by nuzcraft** [comment]
This is awesome!! Do you still have more changes to make, or is this good to merge?

**Comment by felipebrunetti94** [comment]
I was thinking in a visual update, like: "out of spells" color in read or blinking, strong spells in bold, things like that...if you are okay with it I can make the changes, or we can merge and open another issue, for a visual update.

**Comment by felipebrunetti94** [comment]
@nuzcraft  Now it is ready for merge :D

**Pull request merged**